Python/optimizer_bytecodes.c

```diff
@@ -422,6 +425,7 @@ dummy_func(void) {
 
     op(_COMPARE_OP, (left, right -- res)) {
         if (oparg & 16) {
+            REPLACE_OPCODE_IF_EVALUATES_PURE(left, right);
```

Comment:
This will do the right thing even in contexts where the result isn't coerced to bool (all of the cases we care about will produce one anyways):

```suggestion
        REPLACE_OPCODE_IF_EVALUATES_PURE(left, right);
        if (oparg & 16) {
```